tests/ha/clvm.pm

```diff
@@ -8,84 +8,104 @@
 # without any warranty.
 
 # Summary: Create clustered LVM in HA tests
-# Maintainer: Denis Zyuzin <dzyuzin@suse.com>
+# Based on script by Denis Zyuzin
+# Maintainer: Loic Devulder <ldevulder@suse.com>
 
-use base "hacluster";
+use base 'hacluster';
 use strict;
 use testapi;
 use autotest;
 use lockapi;
 
 sub run {
     my $self             = shift;
-    my $vg_name          = "vg_openqa";
-    my $lv_name          = "lv_openqa";
-    my $clvm_partition_1 = "/dev/disk/by-path/ip-*-lun-4";
-    my $clvm_partition_2 = "/dev/disk/by-path/ip-*-lun-5";
-
-    barrier_wait("CLVM_INIT_" . $self->cluster_name);
-    # assert_script_run "zypper -n install dlm-kmp-default lvm2-clvm lvm2-cmirrord";
-    assert_script_run q(sed -ie '/^ *filter/d' /etc/lvm/lvm.conf);                           #by default /dev/disk/by-path/ is filtered in lvm.conf
-    assert_script_run q(sed -ie 's/^\\( *use_lvmetad *=\\) *1/\1 0/' /etc/lvm/lvm.conf);     #set use_lvmetad = 0, lvmetad doesn't support cLVM
-    assert_script_run q(sed -ie 's/^\\( *locking_type *=\\) *1/\1 3/' /etc/lvm/lvm.conf);    #use locking_type=3 for cLVM
+    my $lvm_conf         = '/etc/lvm/lvm.conf';
+    my $vg_name          = 'vg_openqa';
+    my $lv_name          = 'lv_openqa';
+    my $clvm_partition_1 = '/dev/disk/by-path/ip-*-lun-1';
+    my $clvm_partition_2 = '/dev/disk/by-path/ip-*-lun-2';
+
+    # Wait until cLVM test is initialized
+    barrier_wait('CLVM_INIT_' . $self->cluster_name);
+
+    # Test if cLVM package is installed
+    assert_script_run q(rpm -q lvm2-clvm lvm2-cmirrord);
+
+    # Configure LVN for HA cluster
+    assert_script_run qq(sed -ie '/^ *filter/d' $lvm_conf);                            # By default /dev/disk/by-path/ is filtered in lvm.conf
+    assert_script_run qq(sed -ie 's/^\\( *use_lvmetad *=\\) *1/\\1 0/' $lvm_conf);     # Set use_lvmetad = 0, lvmetad doesn't support cLVM
+    assert_script_run qq(sed -ie 's/^\\( *locking_type *=\\) *1/\\1 3/' $lvm_conf);    # Use locking_type=3 for cLVM
     if ($self->is_node1) {
-        type_string "echo wait until clvmd resource is created\n";
+        diag q(Wait until clvmd resource is created...);
```

Comment:
every type_string is also repeated in the logfile so that shouldn't harm. of course diag can be used.